### ISSUE TITLE
Create custom model for 230.000 images

### ISSUE THREAD

**Issue by ogulcankarayel5** [AlexeyAB/darknet#4350]
I have jpg files that contains 230.000 car plate images for graduate project and i'm new for this topic.
What exactly should I do for plate recognation. Which format should I use and how can I convert 230.000 images to data format that you will say.Also do i have to convert images files with my own hands or is there any tool for 230.000 images.

**Comment by AlexeyAB** [comment]
1. How to label your dataset: https://github.com/AlexeyAB/darknet#how-to-mark-bounded-boxes-of-objects-and-create-annotation-files

in C++: https://github.com/AlexeyAB/Yolo_mark
in Python: https://github.com/tzutalin/labelImg
in Python: https://github.com/Cartucho/OpenLabeling
in C++: https://www.ccoderun.ca/darkmark/


2. How to train (to detect your custom objects): https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

Just use `yolov3-spp.cfg` as base cfg.

----

Just to know, Yolo is object Detector, so it can detect license plate and separate letters with its coordinates, but you should collect from them the license plate number yourself in your code.

**Comment by ogulcankarayel5** [comment]
Thanks but I wonder something.It's impossible to convert 230.000 images with hands so do you have any suggestion ?

**Comment by AlexeyAB** [comment]
You shouldn't convert images.

**Comment by ogulcankarayel5** [comment]
Sorry,but i'm very confused.Don't we need annotation for each image

**Comment by AlexeyAB** [comment]
Yes, you must create txt-files with annotations.
But jpg-images shouldn't be changed.

**Comment by ogulcankarayel5** [comment]
Yes but how can I create txt files with annotations each 230.000 images :) do you have any suggestion. By the way thanks for your answers,if my question is bothering you just sorry because they are a little ridicilous I guess , but I'm new on this topic .

**Comment by AlexeyAB** [comment]
If you already have annotation in another format - then create script for converting.
Othewise you should label objects manually.

**Comment by ogulcankarayel5** [comment]
Do you have a tool for labeling

**Comment by ogulcankarayel5** [comment]
Can i use this site for annotation:
https://supervise.ly

**Comment by AlexeyAB** [comment]
I don't know, try to use it. And tell us about the results.
What can they do and for what money?

**Comment by ogulcankarayel5** [comment]
I guess a web platform for deep learning things and priceless. If i use it, of course I tell the results.

**Comment by LukeAI** [comment]
@ogulcankarayel5  which dataset is it? I might be able to help

**Comment by ogulcankarayel5** [comment]
@LukeAI just images right now but i try to annotate to yolo format with supervisely

**Comment by LukeAI** [comment]
but where did you get the images? Is it a private dataset or public?

**Comment by ogulcankarayel5** [comment]
I got images from my teacher for graduate project so it is private i can't share all of it. But i can share a little

**Comment by LukeAI** [comment]
ok. It will take you forever to annotate 250,000 images - my suggestion would be to train a model on an already labelled dataset (try looking here https://platerecognizer.com/number-plate-datasets/) and using that model to label your data for you. Then manually reviewing the generated labels. You can click through much faster and just make corrections as necessary.
I recommend CVAT for annotating - it can import and export in Yolo format too.

**Comment by AlexeyAB** [comment]
@LukeAI Is https://github.com/opencv/cvat better than these? May be I will add URL to it: https://github.com/AlexeyAB/darknet#how-to-mark-bounded-boxes-of-objects-and-create-annotation-files

in C++: https://github.com/AlexeyAB/Yolo_mark
in Python: https://github.com/tzutalin/labelImg
in Python: https://github.com/Cartucho/OpenLabeling
in C++: https://www.ccoderun.ca/darkmark/

**Comment by AlexeyAB** [comment]
@ogulcankarayel5 Yes, you can label 1000 images from your dataset, then train the yolov3-spp.cfg model, and then run detection on other 230 000 images, so your trained model will detect objects on other images and will label these objects - by using pseudo-labeling:

> Pseudo-lableing - to process a list of images data/new_train.txt and save results of detection in Yolo training format for each image as label <image_name>.txt (in this way you can increase the amount of training data) use: `./darknet detector test cfg/coco.data cfg/yolov3.cfg yolov3.weights -thresh 0.25 -dont_show -save_labels < data/new_train.txt`

Then you can just run one of these tools, check labels if they are correct, and fix only incorrect labels:
in C++: https://github.com/AlexeyAB/Yolo_mark
in Python: https://github.com/tzutalin/labelImg
in Python: https://github.com/Cartucho/OpenLabeling
in C++: https://www.ccoderun.ca/darkmark/
JavaScript: https://github.com/opencv/cvat

**Comment by LukeAI** [comment]
> @LukeAI Is https://github.com/opencv/cvat better than these? May be I will add URL to it: https://github.com/AlexeyAB/darknet#how-to-mark-bounded-boxes-of-objects-and-create-annotation-files
> 
> in C++: https://github.com/AlexeyAB/Yolo_mark
> in Python: https://github.com/tzutalin/labelImg
> in Python: https://github.com/Cartucho/OpenLabeling
> in C++: https://www.ccoderun.ca/darkmark/

I found CVAT to be more productive and easier to use than labelimg and yolo_mark, another nice thing is that it runs as a web server so it's easy to create and manage jobs centrally and give them to lots of others.

**Comment by ogulcankarayel5** [comment]
Thanks both of you ,i will try that you suggested and tell the results

**Comment by ogulcankarayel5** [comment]
Hi again,
0 0.496875 0.583333333333 0.1390625 0.113888888889
Is this the correct annotating for this repo.

**Comment by AlexeyAB** [comment]
For Yolo - yes.
For your objects - I dont know.

**Comment by ogulcankarayel5** [comment]
This annotation is my image annotation that I've just created using tool that you sent me.
